core/src/queue.rs

```diff
@@ -115,6 +117,37 @@ impl Queue {
             .collect()
     }
 
+    /// Returns latest asset registering transaction hash.
+    pub fn latest_asset_reg_tx_hash(
+        &self,
+        wsv: &WorldStateView,
+    ) -> Option<HashOf<VersionedSignedTransaction>> {
+        self.txs
+            .iter()
+            .filter(|e| -> bool {
+                self.is_pending(e.value(), wsv)
```

Comment:
Are you sure this transaction is guaranteed to be the one that is being part of the block in the current sumeragi round?

Reply:
To make the match more precise, added another argument to match against the `AssetDefinitionId`.

Reply:
that still might not be correct, you can have two transactions registering the same asset in the queue and you don't know which one is the one that is being voted for in the current consensus round. I'd like to hear other opinions